packages/suppression_check.py

```diff
@@ -246,53 +251,72 @@ def redact_threshold(self):
-    #     self.df_log.loc[mask, 'RedactBinary'] = 1
+        # Grouping by Organization and counting StudentCount, then filtering groups with a single record 
+        if self.organization_columns[0] is not None: 
+            for sensitive_combination in self.sensitive_combinations:
+                list_combination = list(sensitive_combination)
+                if list_combination != self.sensitive_columns:
+                    string_combination = ''.join(list_combination)
+                    df_redact_less = df_log_na[df_log_na['RedactBinary'] == 1]
+                    df_redact_less.loc[:, 'Redacted'] = 1
+                    df_count = df_redact_less.groupby(['Grouping'] + self.organization_columns + list_combination)['Redacted'].sum().reset_index()
+                    df_summed_less_than = df_summed_value[df_summed_value[self.frequency] <= self.minimum_threshold]
+                    if not df_summed_less_than.empty:
+                        df_not_redacted = df_log_na[df_log_na['RedactBinary'] != 1]
+                        df_minimum = df_not_redacted.groupby(['Grouping'] + self.organization_columns + list_combination, dropna=False)[self.frequency].min().reset_index()
+                        df_minimum.rename(columns={self.frequency:'LastMiniumValue'}, inplace=True)
+                        df_minimum_redacted = df_summed_less_than.merge(df_minimum, on = ['Grouping'] + self.organization_columns + list_combination)
+                        df_minimum_redacted = df_minimum_redacted[['Grouping'] + self.organization_columns + list_combination + ['LastMiniumValue']]
+                        df_minimum_one = df_log_na.merge(df_minimum_redacted, on = ['Grouping'] + self.organization_columns + list_combination, how='left')
+                        condition = (df_minimum_one[self.frequency] == df_minimum_one['LastMiniumValue'])
+                        df_log_na.loc[condition, 'RedactBinary'] = 1
+                        df_log_na.loc[condition, 'Redact'] = 'Secondary Suppression'
+                        df_log_na.loc[condition, 'RedactBreakdown'] += ', Sum of values less than threshold'
         
-    #     # Update the 'Redact' column with a specific message for rows where 'MinimumValue' is not null
-    #     self.df_log.loc[mask, 'Redact'] = 'Secondary Suppression'
-    #     #self.df_log.loc[mask, 'Redact'] = 'Sum of minimum threshold redact needed secondary suppression'
+        else:
+            for sensitive_combination in self.sensitive_combinations:
+                list_combination = list(sensitive_combination)
+                if list_combination != self.sensitive_columns:
+                    string_combination = ''.join(list_combination)
+                    df_redact_less = df_log_na[df_log_na['RedactBinary'] == 1]
+                    df_redact_less.loc[:, 'Redacted'] = 1
+                    df_summed_value = df_redact_less.groupby(['Grouping'] + list_combination, dropna=False)[self.frequency].sum().reset_index()
```

Comment:
Looks like df_summed_value is assigned here, so the bug above will only show up when organization columns are specified.